src/production.cpp

```diff
@@ -59,22 +63,36 @@ static void readTagsAndRunConveyor(TaskContext *ctx) {
         }
 
         if (tagReader.isNewTagPresent()) {
-          unsigned char buffer[10];
-          unsigned char size = tagReader.readTag(buffer);
-          if (size > 0) {
-            tag = "";
-            for (unsigned char i = 0; i < size; i++) {
-              char two[3];
-              sniprintf(two, sizeof(two), "%02x", buffer[i]);
-              tag += two;
+          const int currentTick = millis();
+          LOG_TRACE("[TAG] lastTagReadTick %u ; currentTick %u\n", lastTagReadTick, currentTick);
+
+          if (lastTagReadTick + TAG_READER_INTERVAL * 2 <= currentTick) {
+            consecutiveTagRead = 0;
+
+            unsigned char buffer[10];
+            unsigned char size = tagReader.readTag(buffer);
+            if (size > 0) {
+              tag = "";
+              for (unsigned char i = 0; i < size; i++) {
+                char two[3];
+                sniprintf(two, sizeof(two), "%02x", buffer[i]);
+                tag += two;
+              }
+              taskENTER_CRITICAL(&values->subTaskLock);
+              base10tag = strtol(tag.c_str(), &endptr, 16);
+              values->tags.push(&base10tag);
+              taskEXIT_CRITICAL(&values->subTaskLock);
+
+              LOG_INFO("[TAG] New Tag %s\n", tag.c_str());
+            }
+          } else {
```

Comment:
Add a `tagReader.isStuck()` condition to that `else` branch, otherwise the logs are flooded until the next self test.

```
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] A product is stuck!
[E] [TAG] Self test failed
```